internal/mock/cloudsql.go

```diff
@@ -63,6 +69,54 @@ func WithPublicIP(addr string) FakeCSQLInstanceOption {
 	}
 }
 
+// WithPrivateIP sets the private IP address to addr.
+func WithPrivateIP(addr string) FakeCSQLInstanceOption {
+	return func(f *FakeCSQLInstance) {
+		f.ipAddrs["PRIVATE"] = addr
+	}
+}
+
+// WithCertExpiry sets the server certificate's expiration to t.
+func WithCertExpiry(t time.Time) FakeCSQLInstanceOption {
+	return func(f *FakeCSQLInstance) {
+		f.Cert.NotAfter = t
+	}
+}
+
+// WithRegion sets the server's region to the provided value.
+func WithRegion(region string) FakeCSQLInstanceOption {
+	return func(f *FakeCSQLInstance) {
+		f.region = region
+	}
+}
+
+// WithFirstGenBackend sets the server backend type to FIRST_GEN.
+func WithFirstGenBackend() FakeCSQLInstanceOption {
+	return func(f *FakeCSQLInstance) {
+		f.backendType = "FIRST_GEN"
+	}
+}
+
+// SignFunc is a function that signs the certificate using the provided key. The
+// result should be PEM-encoded.
+type SignFunc = func(*x509.Certificate, *rsa.PrivateKey) ([]byte, error)
+
+// WithSigner configures the signing function used to generate a signed
+// certificate.
+func WithSigner(s SignFunc) FakeCSQLInstanceOption {
+	return func(f *FakeCSQLInstance) {
+		f.sign = s
+	}
+}
```

Comment:
nit: This seems like it would be more understandable as "WithCertSigner"

Reply:
Agreed.